lib/kitchen/driver/azure_credentials.rb

```diff
@@ -38,33 +38,49 @@ def initialize(subscription_id:, environment: "Azure")
       # @return [Object] Object that can be supplied along with all Azure client requests.
       #
       def azure_options
-        options = { tenant_id: tenant_id,
-                    client_id: client_id,
-                    client_secret: client_secret,
+        options = { tenant_id: tenant_id!,
                     subscription_id: subscription_id,
                     credentials: ::MsRest::TokenCredentials.new(token_provider),
                     active_directory_settings: ad_settings,
                     base_url: endpoint_settings.resource_manager_endpoint_url }
-
+        options[:client_id] = client_id if client_id
+        options[:client_secret] = client_secret if client_secret
         options
       end
 
       private
```

Comment:
```suggestion
      private
      
```
There should be a space before and after `private`.